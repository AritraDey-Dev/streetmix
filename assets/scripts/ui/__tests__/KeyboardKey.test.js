/* eslint-env jest */
import React from 'react'
import { cleanup } from '@testing-library/react'
import KeyboardKey from '../KeyboardKey'
import { renderWithReduxAndIntl } from '../../../../test/helpers/render'

describe('KeyboardKey', () => {
  it('renders a <kbd> element with string child', () => {
    const wrapper = renderWithReduxAndIntl(
      <KeyboardKey title="fo">foo</KeyboardKey>
    )
    expect(wrapper.getByText('foo').title).toBe('')
    expect(wrapper.asFragment()).toMatchSnapshot()
  })

  it('renders a <kbd> element with element child', () => {
    const wrapper = renderWithReduxAndIntl(
      <KeyboardKey>
        <strong>foo</strong>
      </KeyboardKey>
    )
    expect(wrapper.getByText('foo').title).toBe('')
    expect(wrapper.asFragment()).toMatchSnapshot()
  })

  it('renders a <kbd> element with icon and title', async () => {
    const wrapper = renderWithReduxAndIntl(
      <KeyboardKey icon={{ prefix: 'fas', iconName: 'minus' }}>foo</KeyboardKey>
    )

    const elementWithTitle = await wrapper.getByTitle('foo')
    const iconElement = wrapper.container.querySelector('svg')

    expect(elementWithTitle).toBeDefined()
    expect(elementWithTitle.children).toHaveLength(1)
    expect(iconElement.children).toBeDefined()
    expect(wrapper.asFragment()).toMatchSnapshot()
  })
})
