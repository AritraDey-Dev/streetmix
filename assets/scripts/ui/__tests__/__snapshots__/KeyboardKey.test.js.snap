// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`KeyboardKey renders a <kbd> element with element child 1`] = `
"<kbd className=\\"key\\">
  <strong>
    foo
  </strong>
</kbd>"
`;

exports[`KeyboardKey renders a <kbd> element with icon and title 1`] = `
"<kbd className=\\"key key-icon\\" title=\\"foo\\">
  <FontAwesomeIcon icon={{...}} border={false} className=\\"\\" mask={{...}} fixedWidth={false} inverse={false} flip={{...}} listItem={false} pull={{...}} pulse={false} rotation={{...}} size={{...}} spin={false} symbol={false} title=\\"\\" transform={{...}} />
</kbd>"
`;

exports[`KeyboardKey renders a <kbd> element with string child 1`] = `
"<kbd className=\\"key\\">
  foo
</kbd>"
`;
